### ISSUE TITLE
QtHelp should generate identifiers for keywords in qhp files.

### ISSUE THREAD

**Issue by shimizukawa** [sphinx-doc/sphinx#1491]
This is necessary for the http://doc-snapshot.qt-project.org/qt5-5.3/qhelpenginecore.html#linksForIdentifier API to work. 

Qt-generated qhp files include a id attribute in the keyword element. In CMake, we would need to have 

```
id="${domain_object_name}/${name}"
```

such as

```
id="command/add_library"
id="prop_tgt/COMPILE_DEFINITIONS"
```

---
- Bitbucket: https://bitbucket.org/birkenfeld/sphinx/issue/1491
- Originally reported by: Anonymous
- Originally created at: 2014-06-17T11:06:14.433

**Comment by polwel** [comment]
I also just hit this problem. I need Qt's `linksForIdentifier` to work, yet the `id` field is not populated in the `qhp` files. I see that I can manually add an index entry in the `.rst` files like so

```
.. index:: NAME (ID)
```

However, this will generate `name="NAME (ID)" id="NAME.ID"`, so there are two things I don't like.
1. I cannot specify an entry that does not have a `name`, so it will always show in the index.
2. I don't have full control over the `id`, as it is always prepended with `NAME .`.

Ideally I'd like to be able to specify both the name and the id independently.

**Comment by kreuzberger** [comment]
I also ran in this issue. I have a problem with it providing multilingual support.
I have a german and an english rst file. The code from where i target the qthelp should be using id's only.
Cause id contains name, the name is language specific. This breaks this concept.
The id should be the id only, the name should be the name only

**Comment by kreuzberger** [comment]
Added an option to generate those ids. now .. index:: keyword (id) will be seperated if option is enabled.

Then the index has a keyword in the language of the document, the id could be refered from the source code to activate the item in the manual.
Hopefully integrated soon :+1:

**Comment by kreuzberger** [comment]
Closed as solution in #1491 should be implemented by myself

**Comment by AA-Turner** [comment]
Issue should be reported against `sphinxcontrib/qthelp`.

A
